yew/src/html/listener/registry.rs

```diff
@@ -0,0 +1,503 @@
+use crate::virtual_dom::{Listener, Listeners};
+use std::{
+    cell::RefCell,
+    collections::{HashMap, HashSet},
+    rc::Rc,
+};
+use wasm_bindgen::{prelude::*, JsCast};
+use web_sys::{Element, Event};
+
+thread_local! {
+    /// Global event listener registry
+    static REGISTRY: RefCell<Registry> = Default::default();
+
+    /// Key used to store listener id on element
+    static LISTENER_ID_PROP: wasm_bindgen::JsValue = "__yew_listener_id".into();
+
+    static BODY: web_sys::HtmlElement = web_sys::window()
+        .expect("no window global")
+        .document()
+        .expect("no document on window")
```

Comment:
You can use `yew::utils::document()` here. If you want, you can add the more descriptive error messages there